docs/environment-variables.md

```diff
@@ -31,6 +31,7 @@ This document outlines environment variables that can be used to customize behav
 | `DISABLE_MAX_RECURSION`                  | `bool`              | `false`                                                                                     | Set to true to disable the recursion preventer, which will stop a workflow running which has called into a child template 100 times                                                                                                                                      |
 | `EXPRESSION_TEMPLATES`                   | `bool`              | `true`                                                                                      | Escape hatch to disable expression templates.                                                                                                                                                                                                                            |
 | `EVENT_AGGREGATION_WITH_ANNOTATIONS`     | `bool`              | `false`                                                                                     | Whether event annotations will be used when aggregating events.                                                                                                                                                                                                          |
+| `EXECUTOR_SIGNAL_TIMEOUT`                | `time.Duration`     | `10m`                                                                                       | When the executor does not finish after this time, it will times out and simply returns an err                                                                                                                                                                           |
```

Comment:
Suggest rewording the help:
`How long to wait after the executor has been terminated before returning an error`

Reply:
Default I think 10 minutes will do, which is a time that will never happen. It's also a very safe time. Thanks

Reply:
I think that's a good idea, Thanks!

Reply:
Follow-up [below](https://github.com/argoproj/argo-workflows/pull/13012#discussion_r1594592937)

> I'm happy with 10 minutes - although @agilgur5 also had [an opinion](https://github.com/argoproj/argo-workflows/pull/13012#discussion_r1592666100) on that.

But yes, if this is solely to prevent blocking, we can set this lower and don't need it to be configurable. If it were to be configurable it might make sense as a duration _after_ `terminationGracePeriodSeconds`.

I imagine 2min or lower would be safe as a default

Reply:
@agilgur5 @Joibel In my opinion, i think the best is 2min too. I think it needs to be configurable, it should add some flexibility.

Reply:
> Because it is often difficult to update code in production environments, it is relatively easy to update configuration

For sure, but you wrote [above](https://github.com/argoproj/argo-workflows/pull/13012#discussion_r1592743019) that this should never be hit, so in that case, it shouldn't need configuring.